### PR TITLE
Incorrred parsing of Azure input variables; ref. issue 4534

### PR DESCRIPTION
**Pull request by krstp**
* [ x ] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.
Cannot run a full test as there are issues with test library dependencies; even if present python refuses to load them. Sorry, no time to toy with it.

* [ x ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.
No documentation update yet; but I am willing to provide guidance whenever approved. Whole path forward, I believe, is quite well described in https://github.com/iterative/dvc/issues/4534. Connect with @efiop for detaills.

### CODE REVIEW

File: dvc/tree/azure.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def __init__(self, repo, config):
 
         self._account_url = None
         if not self._conn_str:
-            name = self._az_config.get("storage", "account", None)
+            name = config.get("account")
```
Comment (efiop) [inline dvc/tree/azure.py:45]:
```suggestion
            name = config.get("account") or self._az_config.get("storage", "account", None)
```

File: dvc/tree/azure.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def __init__(self, repo, config):
 
         self._account_url = None
         if not self._conn_str:
-            name = self._az_config.get("storage", "account", None)
+            name = config.get("account") or self._az_config.get("storage", "account", None)
```
Comment (efiop) [inline dvc/tree/azure.py:45]: Btw, I suppose you were testing out things through API? If so, for CLI we'll need to add `account` and `sas_token` to https://github.com/iterative/dvc/blob/67ddb7b41f8ba382048327fa5b5520168ab9c43f/dvc/config.py#L193

File: dvc/tree/azure.py (hunk 1 above)
Comment (krstp) [reply dvc/tree/azure.py]: Sorry; let's revert it back. This code still doesn't work.

File: dvc/tree/azure.py (hunk 2 above)
Comment (krstp) [reply dvc/tree/azure.py:45]: Sorry, this should not have gone through.

**Comment by krstp** [comment]
Closing. Problem persists.

**Pull request closed without merging**